src/components/timeline/ActiveTabResourceTrack.js

```diff
@@ -58,22 +58,26 @@ class ActiveTabResourceTrackComponent extends PureComponent<Props, State> {
     };
   }
 
-  _onMouseUp = (event: MouseEvent) => {
+  _onMouseUp = fromWhere => (event: MouseEvent) => {
     const { isSelected } = this.props;
     const { isOpen } = this.state;
 
     if (event.button === 0) {
       // Don't allow clicks on the threads list to steal focus from the tree view.
       event.preventDefault();
-      if (isSelected || !isOpen) {
-        // We have two different states for selected tracks and open tracks because
-        // non selected tracks also can stay opened. We can only close the track if
-        // the track is already selected.
-        this.setState(prevState => {
-          return { isOpen: !prevState.isOpen };
-        });
+      if (fromWhere === 'timelineTrackResourceLabel') {
```

Comment:
It would be good to add a comment inside of that if branch to make it  easier for people who are going to read it in the future. You can write something like `Only toggle the resource track when the resource label is clicked.`